### PR TITLE
[System.Windows.Forms] Reset check state of item when removing it.

### LINKED ISSUE #7664: CheckListBox maintains check state
## Steps to Reproduce

1. Fill CheckListBox with items: CheckListBox.Items.Add(item,false)
2. Check items in CheckListBox
3. Clear CheckListBox: CheckListBox.Items.Clear()
4. Fill CheckListBox with items: CheckListBox.Items.Add(item,false)
5. Check state from line 2 is maintained even though it was cleared and Item.Count is 0 before step 4.

## Current Behavior
Check state from line 2 is maintained even though it was cleared, items were added expressly setting
check state to false and Item.Count is 0 before step 4.

## Expected Behavior
Items added in step 4 should be unchecked.

### On which platforms did you notice this

[ ] macOS
[ X] Linux (Fedora 27)
[ ] Windows

Same code under Windows 10, VS2017 produces the correct results.

**Version Used**:

Mono version: 5.10.0.160

### Stacktrace
 NA

### PR DESCRIPTION
**Pull request by filipnavara**
Fixes #7664.

### CODE REVIEW

File: mcs/class/System.Windows.Forms/System.Windows.Forms/CheckedListBox.cs [hunk 1]
```diff
@@ -423,7 +423,18 @@ public int Add (object item, CheckState check)
...
+
+			public override void Clear ()
+			{
+				owner.check_states.Clear ();
+				base.Clear ();
+			}
+
+			internal override void UpdateSelection (int removed_index)
+			{
+				owner.check_states.Remove (this[removed_index]);
```
Comment (akoeplinger) [inline mcs/class/System.Windows.Forms/System.Windows.Forms/CheckedListBox.cs]: don't you need to call `base.UpdateSelection (removed_index)` ?

Comment (filipnavara) [reply mcs/class/System.Windows.Forms/System.Windows.Forms/CheckedListBox.cs]: Eh, of course. I was sure I pushed the version where the line was present.

Comment (filipnavara) [reply mcs/class/System.Windows.Forms/System.Windows.Forms/CheckedListBox.cs]: Weird. One `git push --force` later and it's there.

**Comment by akoeplinger** [comment]
@monojenkins commit apidiff

**Pull request merged**